src/transformers/models/ict/configuration_ict.py

```diff
@@ -0,0 +1,283 @@
+        self.num_residual_blocks = num_residual_blocks
+        self.l1_loss_weight = l1_loss_weight
+        self.style_loss_weight = style_loss_weight
+        self.content_loss_weight = content_loss_weight
+        self.inpaint_adv_loss_weight = inpaint_adv_loss_weight
+        self.gan_loss = gan_loss
+
+
+class ICTTransformerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`ICTTransformer`]. It is used to instantiate an
+    ICT model according to the specified arguments, defining the model architecture. Instantiating a configuration with
+    the defaults will yield a similar configuration to that of the [ICT model trained with the ImageNet
+    dataset](https://huggingface.co/sheonhan/ict-imagenet-256).
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        vocab_size (`int`, *optional*, defaults to 512):
+            Vocabulary size of the ICT model. Defines the number of different tokens that can be represented by the
+            `pixel_values` passed when calling [`ICTTransformer`].
+        hidden_size (`int`, *optional*, defaults to 1024):
+            Dimensionality of the embeddings and hidden states.
+        num_hidden_layers (`int`, *optional*, defaults to 35):
+            Number of hidden layers in the Transformer encoder.
+        num_attention_heads (`int`, *optional*, defaults to 8):
+            Number of attention heads for each attention layer in the Transformer encoder.
+        intermediate_size (`int`, *optional*, defaults to 4096):
+            Dimensionality of the "intermediate" (i.e., feed-forward) layer in the Transformer encoder.
+        activation_function (`str`, *optional*, defaults to `"gelu"`):
+            Activation function (can be one of the activation functions defined in src/transformers/activations.py).
+            Defaults to "quick_gelu".
+        embedding_dropout_prob (`int`, *optional*, defaults to 0.0):
+            The dropout ratio for the embeddings.
+        residual_dropout_prob (`float`, *optional*, defaults to 0.0):
+            The dropout probability for all fully connected layers in the embeddings, encoder, and pooler.
+        attention_probs_dropout_prob (`float`, *optional*, defaults to 0.0):
+            The dropout ratio for the attention probabilities.
+        initializer_range (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        layer_norm_eps (`float`, *optional*, defaults to 1e-12):
+            The epsilon used by the layer normalization layers.
+        image_size (`int`, *optional*, defaults to `1024`):
+            The size (resolution) of each image.
+        num_channels (`int`, *optional*, defaults to `3`):
+            The number of input channels.
```

Comment:
nit: we don't need to put numerical values in code formatting in the docstrings

```suggestion
        image_size (`int`, *optional*, defaults to 1024):
            The size (resolution) of each image.
        num_channels (`int`, *optional*, defaults to 3):
            The number of input channels.
```